### ISSUE TITLE
[question] unexpected component render  when first set different state

### ISSUE THREAD

**Issue by webMasterMrBin** [dai-shi/use-context-selector#105]
i copy minimal demo https://github.com/dai-shi/use-context-selector/blob/main/examples/01_minimal/src/index.js 
  to this https://codesandbox.io/s/async-hooks-rlvvt4?file=/src/App.js  
 1. first i click Count + 1, only Count render, TestBox not render, it's right
 
![image](https://user-images.githubusercontent.com/25171709/226155871-e3a3d90e-29b9-4666-84ac-1d743e833d0a.png)

2. then i change TextBox input value,  log Count render,  seems like Count render too when change TextBox state, this is a bug?  i expected  Count should't render  when change TextBox.
![image](https://user-images.githubusercontent.com/25171709/226155950-2f5554e4-7bed-47b0-a656-7f6228137d09.png)

**Comment by dai-shi** [comment]
#97 and #100 should the same questions.

tldr; you want to use `useEffect` for console.log.

**Comment by webMasterMrBin** [comment]
> 

thank you,  i check these same question, and i find react docs  https://legacy.reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update
this is react default behavior  not  this  good library
i think  i  put  console  in useEffect's  outside, is not 'real' render  even it log something,
so i should put console in useEffect to check a component is render, right?

**Comment by dai-shi** [comment]
It depends on what is "render".
React has a) render phase + b) commit phase.
What you observe is, it enters the render phase, bails out, so it doesn't go on to the commit phase.
The render phase does call render functions. If it's heavy computation, the solution is `useMemo`.

**Comment by webMasterMrBin** [comment]
> It depends on what is "render". React has a) render phase + b) commit phase. What you observe is, it enters the render phase, bails out, so it doesn't go on to the commit phase. The render phase does call render functions. If it's heavy computation, the solution is `useMemo`.

great!  thanks again, now i undestand
